### ISSUE TITLE
Cannot view external tables

### ISSUE THREAD

**Issue by christoph88** [bstruct/vscode-bigquery#21]
Really cool plugin! Nice work.

Unfortunately, I cannot view external tables. Is this a bug or do you know when this functionality would become available?

Thanks!

**Comment by DamiaoCastro** [comment]
Hi @christoph88 , thank you for reaching out. 
Viewing external tables was not implemented together with the view of normal tables because, for some reason, is a different API call. 
But now that I know that there's need for it, will be done as soon as I finish the point that I'm working on now, that is the first version of the language server. 
Will keep you updated

**Comment by christoph88** [comment]
Thanks that would be really great. We are working with quite a lot of external tables mostly linked to GCS or Google sheets.

**Comment by DamiaoCastro** [comment]
Hi @christoph88 , just made public a new version of this extension. The solution should fix the issue with viewing external tables. 
I succeeded to list a table based of a Cloud Storage file, but still having issues with Google Sheets based tables because of authentication. 
Keep getting permission issues when trying to read Google Drive
![image](https://user-images.githubusercontent.com/16340298/188258377-25535797-3a36-44e7-9301-9fc2cf2d8745.png)

Even after executing `gcloud auth login --enable-gdrive-access` 
I close this issue, I will open a new one to revisit this entire login setup. It needs to be more flexible that it is now.
Also see that authentication is the main issue that people get stuck on after installing the extension. I will tackle this now.

**Comment by christoph88** [comment]
Hi @DamiaoCastro did you try sharing the file in drive with the same email adress you use for your GC credentials?

Either way thank you for your quick follow up and for the response!

**Comment by DamiaoCastro** [comment]
@christoph88 , found what was missing in my authentication request.
with the command:
```
gcloud auth login --update-adc --add-quota-project-to-adc --quiet --enable-gdrive-access --verbosity warning --format="json"
```
now I can see the tables that are gdrive based. 
Please run this command to authenticate.
I will make changes to the extension to support this.
Thank you

**Comment by christoph88** [comment]
@DamiaoCastro 
I tried running above command in the command line.
I first updated your plugin to v0.0.10 and reloaded vscode.
It did not work.

Then I then revoked my account in vscode and tried to add it again using user login.

But I keep getting the auth errors that I do not have access to Google drive.

Could it be that I am missing something?

**Comment by DamiaoCastro** [comment]
@christoph88  Sorry to hear. I will continue to pursue a better login and credentials setup.
Quick question: Can you open a table based on google cloud storage?
Quick favor: Can please try to run a query ( on a .bqsql file or selecting the BQ language on a new file ) run a simple query on the table that you're having issues, like "SELECT * FROM `your-table` LIMIT 100", and tell me if you have the same error.

Thank you.

**Comment by christoph88** [comment]
@DamiaoCastro thank you for your efforts.

I created a table based on GCS and I can view it even if it is set to external. So no issues there.

Favor: I did as requested but I keep getting that error. Within big query that query works as expected.
![image](https://user-images.githubusercontent.com/7599918/188795841-7123c407-9dc3-4b11-ae17-f2751265927c.png)

**Comment by christoph88** [comment]
I also tried running ```bq query < temp.bsql``` that at first did not work. Running ```gcloud auth login --enable-gdrive-access```again and then in the terminal I could query again.

Tried it in your plugin same error. Revoked access and connected again and keep getting above problem.

**Comment by DamiaoCastro** [comment]
@christoph88 , just released version 0.0.11 that gives visibility on the gcloud commands that run in support of the authentication functionality. Also put it more reset moments of the BQ client to assure that it doesn't hold on to the previous credentials.
Hope this helps.

**Comment by christoph88** [comment]
Hi @DamiaoCastro It seems to work now! I am able to query tables based on g drive.

I still have some issues when combining my work and private account. It looks like it does not switch credentials then properly. Even when activating the proper account. There are also some issues with the explorer in that case.

Either way it's not that big of a deal to revoke an account and add the other one so if you have other features you would like to focus on first I understand. 

Thank you very much for fixing it and for your clear communication. If you ever need my help just let me know I am willing to test.
